src/v3/obfuscate.test.ts

```diff
@@ -0,0 +1,180 @@
+import { obfuscateDocument, obfuscateData } from "./obfuscate";
+import { OpenAttestationCredentialWithInnerIssuer, validateSchema, verifySignature, wrapDocument } from "../";
+import { get } from "lodash";
+
+import { SchemaId } from "../shared/@types/document";
+import { IdentityProofType, Method, OaProofType, TemplateType } from "../../src/__generated__/schemaV3";
+import { toBuffer } from "../shared/utils";
+
+const openAttestationData: OpenAttestationCredentialWithInnerIssuer = {
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://www.w3.org/2018/credentials/examples/v1",
+    "https://nebulis.github.io/tmp-jsonld/OpenAttestation.v3.jsonld",
+    "https://nebulis.github.io/tmp-jsonld/CustomContext.jsonld"
+  ],
+  reference: "document identifier",
+  validFrom: "2010-01-01T19:23:24Z",
+  issuanceDate: "2010-01-01T19:23:24Z",
+  name: "document owner name",
+  type: ["VerifiableCredential", "UniversityDegreeCredential"],
+  credentialSubject: {
+    id: "did:example:ebfeb1f712ebc6f1c276e12ec21",
+    degree: {
+      type: "BachelorDegree",
+      name: "Bachelor of Science in Mechanical Engineering"
+    }
+  },
+  template: {
+    name: "any",
+    type: TemplateType.EmbeddedRenderer,
+    url: "http://some.example.com"
+  },
+  issuer: {
+    id: "http://some.example.com",
+    name: "DEMO STORE",
+    identityProof: {
+      type: IdentityProofType.DNSTxt,
+      location: "tradetrust.io"
+    }
+  },
+  oaProof: {
+    type: OaProofType.OpenAttestationProofMethod,
+    value: "0x9178F546D3FF57D7A6352bD61B80cCCD46199C2d",
+    method: Method.TokenRegistry
+  }
+};
+
+describe("privacy", () => {
+  describe("obfuscateData", () => {
+    test("removes one field", async () => {
+      const testData = {
+        key1: "value1",
+        key2: "value2",
+        ...openAttestationData
+      };
+      const field = "key1";
+      const newDocument = await wrapDocument(testData, { version: SchemaId.v3 });
+      const { obfuscatedData } = obfuscateData(newDocument, field);
+      const salt = newDocument.proof.salts[1];
+      const value = get(newDocument, field);
+      expect(obfuscatedData).toEqual([toBuffer(`${salt.value}:${value}`).toString("hex")]);
```

Comment:
I think there are missing assertions
- the field is not in the data returned
- the salt has been removed